archive/indexer.go

```diff
@@ -8,21 +8,50 @@ import (
 
 	"github.com/brimsec/zq/zbuf"
 	"github.com/brimsec/zq/zdx"
-	"github.com/brimsec/zq/zio/detector"
+	"github.com/brimsec/zq/zio/bzngio"
 	"github.com/brimsec/zq/zng"
 	"github.com/brimsec/zq/zng/resolver"
 )
 
 const zarExt = ".zar"
 
-// TBD
+// XXX Embedding the type and field names like this can result in some clunky
+// file names. We might want to re-work the naming scheme.
+
+func typeZdxName(t zng.Type) string {
+	return "zdx:type:" + t.String()
+}
+
+func fieldZdxName(fieldname string) string {
+	return "zdx:field:" + fieldname
+}
+
+func archiveDir(path string) (string, error) {
+	//XXX for now the index directory is the name of the zng file
```

Comment:
That would be much better.  I didn't put much thought on this and @alfred-landrum has been working on the "big picture" here for how everything looks in the file system.

Reply:
Cool. One day not too far off I'm guessing this will also want to be all be http-capable so that this can work with local indexes and cloud data, or everything in the cloud.

Reply:
Yes, @alfred-landrum has been focused on this future proofing.

Reply:
👍